clang/test/ClangScanDeps/modules-include-tree-implementation-transitive.c

```diff
@@ -0,0 +1,66 @@
+// REQUIRES: ondisk_cas
+
+// This test checks that we correctly handle situations where a spurious modular
+// dependency (1) turns otherwise textual dependency into modular (2).
+//
+// (1) For example #include <Spurious/Missing.h>, where the framework Spurious
+// has an umbrella header that does not include Missing.h, making it a textual
+// include instead.
+//
+// (2) For example when compiling the implementation file of a module Mod,
+// #included headers belonging to Mod are treated textually, unless some other
+// module already depends on Mod in its modular form.
+
+// RUN: rm -rf %t
+// RUN: split-file %s %t
+// RUN: sed "s|DIR|%/t|g" %t/cdb.json.in > %t/cdb.json
+
+//--- cdb.json.in
+[{
+  "file": "DIR/tu.m",
+  "directory": "DIR",
+  "command": "clang -c DIR/tu.m -o DIR/tu.o -F DIR/frameworks -I DIR/include -fmodule-name=Mod -fmodules -fimplicit-module-maps -fmodules-cache-path=DIR/module-cache"
+}]
+
+//--- frameworks/Spurious.framework/Modules/module.modulemap
+framework module Spurious {
+  umbrella header "Spurious.h"
+  module * { export * }
+}
+//--- frameworks/Spurious.framework/Headers/Spurious.h
+#include <Mod.h>
+//--- frameworks/Spurious.framework/Headers/Missing.h
+
+//--- include/module.modulemap
+module Mod { header "Mod.h" }
+//--- include/Mod.h
+typedef int mod_int;
+
+//--- tu.m
+#include <Spurious/Missing.h>
+#include <Mod.h>
```

Comment:
Since the above include just textually includes "Missing.h", just marking Mod as visible here doesn't do anything. We need to either enter it textually, or import it properly (like we do with implicit modules and non-caching explicit modules).